src/utils/stockDataUtils/createStockDataEntry.test.js

```diff
@@ -0,0 +1,258 @@
+        optionalValueFormat: '0.00%'
+      });
+  });
+  it('creates correct data entry given value ' +
+     'and an option that doesn\'t exist', () => {
+    expect(createStockDataEntry(15, { badOption: 'hello' }))
+      .to
+      .deep
+      .equal({
+        value: 15,
+        optionalValue: undefined,
+        valueFormat: NUMBER_FORMAT_DEFAULT,
+        optionalValueFormat: NUMBER_FORMAT_DEFAULT
+      });
+  });
+  it('throws an error if value is letters followed by numbers', () => {
+    expect(() => {
+      createStockDataEntry('hello15');
+    }).to
+      .throw();
+  });
+  it('throws an error if value is numbers followed by letters', () => {
+    expect(() => {
+      createStockDataEntry('15hello');
+    }).to
+      .throw();
+  });
+  it('throws an error if value is a non-numeric string', () => {
+    expect(() => {
+      createStockDataEntry('hello');
+    }).to
+      .throw();
+  });
+  it('throws an error if value is an object', () => {
+    expect(() => {
+      createStockDataEntry({});
+    }).to
+      .throw();
+  });
+  it('throws an error if value is an array', () => {
+    expect(() => {
+      createStockDataEntry([]);
+    }).to
+      .throw();
+  });
+  it('throws an error if value is a boolean', () => {
+    expect(() => {
+      createStockDataEntry(true);
+    }).to
+      .throw();
+  });
+  it('throws an error if value is null', () => {
+    expect(() => {
+      createStockDataEntry(null);
+    }).to
+      .throw();
+  });
+  it('throws an error if value is a function', () => {
+    expect(() => {
+      createStockDataEntry(() => console.log('a function'));
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is letters followed by numbers', () => {
+    expect(() => {
+      createStockDataEntry(15, { optionalValue: 'hello12' });
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is numbers followed by letters', () => {
+    expect(() => {
+      createStockDataEntry(15, { optionalValue: '12hello' });
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is a non-numeric string', () => {
+    expect(() => {
+      createStockDataEntry(15, { optionalValue: 'hello' });
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is an object', () => {
+    expect(() => {
+      createStockDataEntry(15, { optionalValue: {} });
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is an array', () => {
+    expect(() => {
+      createStockDataEntry(15, { optionalValue: [] });
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is a boolean', () => {
+    expect(() => {
+      createStockDataEntry(15, { optionalValue: true });
+    }).to
+      .throw();
+  });
+  it('throws an error if optionalValue is a function', () => {
+    expect(() => {
+      createStockDataEntry(
+        15,
+        { optionalValue: () => console.log('a function') }
```

Comment:
An even dumber function that I use in testing myself is `() => {/* no-op */}`.  If you use this you never have to worry about random console prints or other side effects